### ISSUE TITLE
Make an option to have a random string after the SSID

### ISSUE THREAD

**Issue by diegosucaria** [resin-io/resin-wifi-connect#67]
Adding a random string after the SSID gives you the ability to turn on several devices at the same time, and connect and config one by one.

Add **Math.random().toString(36).slice(4).substring(0, 4)** to the SSID

So, for example, the line #2 at config.coffe will read:
`ssid: process.env.PORTAL_SSID or 'resinAP - ' + Math.random().toString(36).slice(4).substring(0, 4)`

We have just tried it and works good.

Maybe you can do it an option.

**Comment by dhoney** [comment]
I do this in my `start.sh` by grabbing the last 4 of my MAC and putting that in the SSID and PSK like so.

```
check_wifi() {
  export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket

  last4_mac=`cat /sys/class/net/eth0/address | sed 's/://g'`
  last4_mac=`echo $last4_mac | tail -c 5`
  ssid="unprovisioned-device-$last4_mac"
  psk="unprovisioned$last4_mac"

  echo "[STARTUP] Checking WiFi connection manager.."
  /usr/src/app/wifi/resin-wifi-connect --clear=false -i wlan1 -s $ssid -p $psk -u /usr/src/app/wifi/public > /dev/console &
}
```

**Comment by majorz** [comment]
That's a nice one!

**Comment by majorz** [comment]
I think this can be added as a FAQ entry in the README with a gist or a wiki link, since I imagine many people would like to use it. Or maybe even better it should go in the example start script if we can make it independent of the interface name.

**Comment by dhoney** [comment]
That should be fine, interface name can just be passed as an argument to the function like so:

```
check_wifi() {
  export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket

  iface=$1

  last4_mac=`cat /sys/class/net/eth0/address | sed 's/://g'`
  last4_mac=`echo $last4_mac | tail -c 5`
  ssid="unprovisioned-device-$last4_mac"
  psk="unprovisioned$last4_mac"

  echo "[STARTUP] Checking WiFi connection manager.." > /dev/console
  /usr/src/app/wifi/resin-wifi-connect --clear=false -i $iface -s $ssid -p $psk -u /usr/src/app/wifi/public > /dev/console &
}
```

You don't have to declare `$iface` if you don't want to but it helps make it readable.
The example start script should be modified to use V4 too since there should be no `--clear=false` so a check with `iwgetid -r` like in the example script might end up as:

```
check_wifi() {
  export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket

  iface=$1

  last4_mac=`cat /sys/class/net/eth0/address | sed 's/://g'`
  last4_mac=`echo $last4_mac | tail -c 5`
  ssid="unprovisioned-device-$last4_mac"
  psk="device$last4_mac"

  iwgetid -r

  if [ $? -eq 0 ]; then
    echo "[STARTUP] WiFi already connected to: $(iwgetid -r)" > /dev/console
  else
    echo "[STARTUP] Starting WiFi AP" > /dev/console
    /usr/src/app/wifi/wifi-connect -i $iface -s $ssid -p $psk -u /usr/src/app/wifi/ui > /dev/console &
  fi

}
```

**Comment by majorz** [comment]
I was referring to `eth0` from `/sys/class/net/eth0/address`.

**Comment by dhoney** [comment]
Ah yes, that can happen too the same way with argument substitution
It's not guaranteed that every device has an ethernet interface, could be wifi, ble, etc..

```
check_wifi() {
  export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket

  iface=$1

  last4_mac=`cat /sys/class/net/$iface/address | sed 's/://g'`
  last4_mac=`echo $last4_mac | tail -c 5`
  ssid="unprovisioned-device-$last4_mac"
  psk="device$last4_mac"

  iwgetid -r

  if [ $? -eq 0 ]; then
    echo "[STARTUP] WiFi already connected to: $(iwgetid -r)" > /dev/console
  else
    echo "[STARTUP] Starting WiFi AP" > /dev/console
    /usr/src/app/wifi/wifi-connect -i $iface -s $ssid -p $psk -u /usr/src/app/wifi/ui > /dev/console &
  fi

}
```

**Comment by majorz** [comment]
I was recently working on an issue related with mac address randomization during wifi scans. The default behavior under NetworkManager is that when a wifi device is not connected to a network, its mac address will be randomized at each scan. Getting the persistent address (e.g. hardware one) will be challenging from the container.

But we may use `RESIN_DEVICE_UUID` since the start script is for running on top of resinOS anyway. This has the advantage that the uuid is visible on the web dashboard. The first 7 characters are directly displayed there, so my first thought is about a name like "WiFi-Connect-76F64DA". How that does sound to you? Any disadvantages you see?

cc @shaunmulligan @lurch - We are discussing here enhancing the example start.sh script by adding a suffix to the SSID of the captive portal's network. Something that will expand to `./wifi-connect --ssid 'WiFi-Connect-76F64DA'`. Useful for running multiple devices in close range. What do you guys think?

**Comment by dhoney** [comment]
@majorz I really like that approach.
It's nice because it gives context to what device in the management panel is in that state vs a MAC address that is not captured.

I had no idea network manager was doing MAC randomization, seems the reason is [privacy](https://blogs.gnome.org/thaller/2016/08/26/mac-address-spoofing-in-networkmanager-1-4-0/).

This makes me wonder though, even if NM is doing randomization, does it affect on the MAC at the level of calling it from `/sys/class/net/eth0/address` ?

Regarding usefulness, we deploy many devices in single locations so having unique SSIDs is a must which is why we use the last 4 of the MAC.

I'm not sure if the `RESIN_DEVICE_UUID` approach would work for our application because we preload the SD cards (base OS and container)
Is the device UUID part of the preloaded image, or is it given once the device phones home for the first time?
My current understanding is the latter which means if a device comes up with no internet it will not have a UUID, correct?

**Comment by majorz** [comment]
Right, it can even be unmanaged device that will never have resin uuid. I think both options could be provided in the start script. The default can be the uuid one. For the mac address option the user may tweak manually the name of the interface (default being `eth0`).

**Comment by majorz** [comment]
The `/sys/class/net/.../address` will be the same as the randomized one. I could only get the persistent address from the host os with `ethtool --show-permaddr devname`.
